angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -75,7 +75,28 @@ define(['angular', 'jquery'], function(angular, $) {
             ' from your list of favorites, try again later.');
           });
       };
-
+       
+      $rootScope.addPortletToHome = function (fname) {
```

Comment:
:question: could this be factored into a shared service rather than being applied directly to the `$rootScope`?

Reply:
We've got a through rewrite of the layout service coming down the pike. Factoring a new service just for this case would be throwaway work.

Reply:
I was not aware of the rewrite, thanks for the info @Doug-Reed!

Reply:
It's a long story, but we need this in rootScope in order to grab it from within uw-frame. If we put this in a service (like we already have in layoutService), we'd have to inject it where we need to call it in frame, which would put a hard dependency on angularjs-portal from frame.  This way, we can grab it from scope if it's there, and if it's not, no harm.